sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -413,38 +413,36 @@ case class DescribeTableCommand(table: TableIdentifier, isExtended: Boolean, isF
     } else {
       val metadata = catalog.getTableMetadata(table)
 
+      if (DDLUtils.isDatasourceTable(metadata)) {
+        DDLUtils.getSchemaFromTableProperties(metadata) match {
+          case Some(userSpecifiedSchema) => describeSchema(userSpecifiedSchema, result)
+          case None => describeSchema(catalog.lookupRelation(table).schema, result)
+        }
+      } else {
+        describeSchema(metadata.schema, result)
+      }
```

Comment:
@yhuai I just did a try. We have to pass `CatalogTable` for avoiding another call of `getTableMetadata`. We also need to pass `SessionCatalog` for calling `lookupRelation`. Do you like this function? or keep the existing one? Thanks!

```Scala
  private def describeSchema(
      tableDesc: CatalogTable,
      catalog: SessionCatalog,
      buffer: ArrayBuffer[Row]): Unit = {
    if (DDLUtils.isDatasourceTable(tableDesc)) {
      DDLUtils.getSchemaFromTableProperties(tableDesc) match {
        case Some(userSpecifiedSchema) => describeSchema(userSpecifiedSchema, buffer)
        case None => describeSchema(catalog.lookupRelation(table).schema, buffer)
      }
    } else {
      describeSchema(tableDesc.schema, buffer)
    }
  }
```